pymobiledevice3/restore/restore.py

```diff
@@ -59,11 +61,19 @@ def __init__(self, ipsw: zipfile.ZipFile, device: Device, tss=None, behavior: Be
         self._pb_verify_restore = None
         self._pb_verify_restore_old_value = None
 
+        # cache for already downloaded url-assets
+        self._url_assets_cache = {}
+
+        def handle_async_data_request_msg(message: Mapping) -> None:
+            thread = Thread(target=self.handle_data_request_msg, args=(message,))
+            thread.start()
```

Comment:
where this thread join?

Reply:
there is none - it will continue till it gracefully ends
in our point of view, each thread now handles its own completely seprate update logic